pkg/appfile/parser.go

```diff
@@ -773,15 +769,14 @@ func (p *Parser) ValidateComponentNames(app *v1beta1.Application) (int, error) {
 }
 
 // GetScopeDefAndGVK get grouped an API version of the given scope
-func GetScopeDefAndGVK(ctx context.Context, cli client.Reader, dm discoverymapper.DiscoveryMapper,
-	name string) (*v1beta1.ScopeDefinition, metav1.GroupVersionKind, error) {
+func GetScopeDefAndGVK(ctx context.Context, cli client.Client, name string) (*v1beta1.ScopeDefinition, metav1.GroupVersionKind, error) {
 	var gvk metav1.GroupVersionKind
 	sd := new(v1beta1.ScopeDefinition)
 	err := util.GetDefinition(ctx, cli, sd, name)
 	if err != nil {
 		return nil, gvk, err
 	}
-	gvk, err = util.GetGVKFromDefinition(dm, sd.Spec.Reference)
+	gvk, err = util.GetGVKFromDefinition(cli.RESTMapper(), sd.Spec.Reference)
```

Comment:
does the restmapper equal to discovermapper here?

Reply:
Yes